src/modules/integrations/android.nix

```diff
@@ -0,0 +1,270 @@
+      '';
+    };
+
+    includeNDK.enable = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = ''
+        Whether to include the Android NDK (Native Development Kit).
+        By default, the NDK is included.
+      '';
+    };
+
+    useGoogleAPIs.enable = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = ''
+        Whether to use the Google APIs.
+        By default, the Google APIs are used.
+      '';
+    };
+
+    useGoogleTVAddOns.enable = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = ''
+        Whether to use the Google TV Add-Ons.
+        By default, the Google TV Add-Ons are used.
+      '';
+    };
+
+    includeExtras = lib.mkOption {
+      type = lib.types.listOf lib.types.str;
+      default = [ "extras;google;gcm" ];
+      description = ''
+        The Android extras to install.
+        By default, the Google Cloud Messaging (GCM) extra is installed.
+      '';
+    };
+
+    extraLicenses = lib.mkOption {
+      type = lib.types.listOf lib.types.str;
+      default = [
+        "android-sdk-preview-license"
+        "android-googletv-license"
+        "android-sdk-arm-dbt-license"
+        "google-gdk-license"
+        "intel-android-extra-license"
+        "intel-android-sysimage-license"
+        "mips-android-sysimage-license"
+      ];
+      description = ''
+        The additional Android licenses to accept.
+        By default, several standard licenses are accepted.
+      '';
+    };
+
+    jdk = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.jdk;
+      defaultText = "pkgs.jdk";
+      description = ''
+        The JDK package to use.
+        By default, the JDK package from nixpkgs is used.
+      '';
+      example = "pkgs.jdk";
+    };
+
+    android-studio = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.android-studio;
+      defaultText = "pkgs.android-studio";
+      description = ''
+        The Android Studio package to use.
+        By default, the Android Studio package from nixpkgs is used.
+      '';
+      example = "pkgs.android-studio";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    packages = [
+      androidSdk
+      platformTools
+      androidEmulator
+      cfg.jdk
+      cfg.android-studio
+    ];
+
+    env.ANDROID_HOME = "${androidSdk}/libexec/android-sdk";
+    env.ANDROID_NDK_ROOT = "${config.env.ANDROID_HOME}/ndk/";
+
+    # override the aapt2 binary that gradle uses with the patched one from the sdk
+    env.GRADLE_OPTS = "-Dorg.gradle.project.android.aapt2FromMavenOverride=${androidSdk}/libexec/android-sdk/build-tools/${lib.head cfg.buildTools.version}/aapt2";
+
+    enterShell = ''
+      set -e
+      export PATH="$PATH:${config.env.ANDROID_HOME}/tools:${config.env.ANDROID_HOME}/tools/bin:${config.env.ANDROID_HOME}/platform-tools"
+      cat <<EOF > local.properties
```

Comment:
Does this need to live in the root of the project?

Reply:
Yes it does.